neqo-udp/src/lib.rs

```diff
@@ -7,20 +7,66 @@
 #![allow(clippy::missing_errors_doc)] // Functions simply delegate to tokio and quinn-udp.
 
 use std::{
+    array,
     io::{self, IoSliceMut},
+    iter,
     net::SocketAddr,
     slice::{self, Chunks},
 };
 
+use log::{log_enabled, Level};
 use neqo_common::{qdebug, qtrace, Datagram, IpTos};
 use quinn_udp::{EcnCodepoint, RecvMeta, Transmit, UdpSocketState};
 
-/// Socket receive buffer size.
+/// Receive buffer size
 ///
-/// Allows reading multiple datagrams in a single [`Socket::recv`] call.
-//
-// TODO: Experiment with different values across platforms.
-pub const RECV_BUF_SIZE: usize = u16::MAX as usize;
+/// Fits a maximum size UDP datagram, or, on platforms with segmentation
+/// offloading, multiple smaller datagrams.
+const RECV_BUF_SIZE: usize = u16::MAX as usize;
+
+/// The number of buffers to pass to the OS on [`Socket::recv`].
+///
+/// Platforms without segmentation offloading, i.e. platforms not able to read
+/// multiple datagrams into a single buffer, can benefit from using multiple
+/// buffers instead.
+///
+/// Platforms with segmentation offloading don't benefit from additionally using
+/// multiple buffers.
+///
+/// - Linux/Android: use segmentation offloading via GRO
+/// - Windows: use segmentation offloading via URO (caveat see <https://github.com/quinn-rs/quinn/issues/2041>)
+/// - Apple: no segmentation offloading available, use multiple buffers
+#[cfg(not(any(
+    target_os = "macos",
+    target_os = "ios",
+    target_os = "tvos",
+    target_os = "visionos"
+)))]
```

Comment:
Wonder if we should start using `cfg_aliases`, similar to what `quinn` does: https://github.com/quinn-rs/quinn/blob/main/quinn-udp/build.rs

Reply:
Good idea. I will look into it.